Better animation blending & simple animation controller

HollowHorizon opened the pull request:
I rewrote the animation system from global keyframes to local ones so that you can combine as many animations as you want and their priorities.

Also animation controller template:
```kotlin
val animator = AnimationController(model).apply {
    initialState = "Idle"
    state("Idle") {
        animations("idle")
        transition("Walking") { /* Predicate when model starts walking */ }
    }
    state("Walking") {
        animations("walk")
        transition("Idle") { /* Predicate when model stops walking */ }
    }
}
model.onUpdate { animator.update(Time.deltaT) }
```

fabmax reviewed (changes requested):
Pretty nice! I have a few minor nitpicks (see comments) but nothing critical.

HollowHorizon commented:
It sounds quite reasonable, at first I didn't notice the presence of mix functions and out of habit I searched for lerp)
I rewrote everything using mix, and also replaced the simple linear interpolation for quaternions with spherical (for rotations).

Without this, animations look something like this :D

https://github.com/user-attachments/assets/92651611-2b5e-4151-ae48-f6f4b9bc0790

Also I see that you would like to avoid one more loop when playing animations, but honestly I don't know how to do it better. The problem here is that if you move the check to the first loop, and for example there will be two animations using the “left_arm” bone, then before applying the second animation it will reset this bone once again after applying the first one.

fabmax commented on kool-core/src/commonMain/kotlin/de/fabmax/kool/scene/animation/Animation.kt:
```diff
@@ -149,8 +152,8 @@ class AnimatedTransformGroup(val target: Node) : AnimationNode {
         t.translate(animTranslation.mul(weight))
-        t.rotate(MutableQuatF().slerp(animRotation, weight))
-        t.scale(MutableVec3f(Vec3f.ONES).lerp(animScale, weight))
+        t.rotate(baseRotation.mix(animRotation, weight))
```
Now with `baseRotation` and `baseScale` being members, I think it's also needed to reset them before calling mix, otherwise the previous values will influence the outcome. So:

```
t.rotate(baseRotation.setIdentity().mix(animRotation, weight))
t.scale(baseScale.set(Vec3f.ONES).mix(animScale, weight))
```

HollowHorizon commented:
No, why, it writes the result to a new object, not the same one. 
![image](https://github.com/user-attachments/assets/5b628f57-455d-4a21-bd5c-cc2903a6d4c0) 
![image](https://github.com/user-attachments/assets/58ec010d-6a9d-4f11-8557-91cc7bb40ea1)

fabmax commented:
Ah I see, I don't know my own API apprarently 😆 
I guess the receivers wouldn't need to be mutable then. But anyway it's fine.
